OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -101,13 +101,36 @@ public class BulletInfo : IProjectileInfo
 		[Desc("Altitude above terrain below which to explode. Zero effectively deactivates airburst.")]
 		public readonly WDist AirburstAltitude = WDist.Zero;
 
+		[Desc("Length of the contrail (in ticks).")]
 		public readonly int ContrailLength = 0;
-		public readonly int ContrailZOffset = 2047;
-		public readonly Color ContrailColor = Color.White;
-		public readonly bool ContrailUsePlayerColor = false;
+
+		[Desc("Delay of the contrail (in ticks).")]
 		public readonly int ContrailDelay = 1;
+
+		[Desc("Offset for contrail's Z sorting.")]
+		public readonly int ContrailZOffset = 2047;
+
+		[Desc("Width of the contrail.")]
 		public readonly WDist ContrailWidth = new WDist(64);
 
+		[Desc("RGB color at the contrail start.")]
+		public readonly Color ContrailStartColor = Color.White;
+
+		[Desc("Use player remap color instead of a custom color at the contrail the start.")]
+		public readonly bool ContrailStartColorUsePlayerColor = false;
+
+		[Desc("The alpha value [from 0 to 255] of color at the contrail the start.")]
+		public readonly int ContrailStartColorAlpha = 255;
```

Comment:
You don't need this.

> Color (RRGGBB[AA] notation)

This can be defined in `ContrailStartColor`.

Reply:
I am affraid we still need this, considering we can also set up player color and we want to add an alpha control to it

Reply:
For example, we fade with player color but alpha=0

Reply:
I see.